dovi_tool/entrypoint.sh

```diff
@@ -65,26 +60,73 @@ get_dvhe_profile() {
 	fi
 }
 
-# Demux the file using ffmpeg and dovi_tool
-demux_file() {
-	echo "Demuxing $1..."
+extract_mkv() {
+	echo "Extracting $1..."
 	echo "------------------"
-	echo "ffmpeg -i $1 -c:v copy -vbsf hevc_mp4toannexb -f hevc - | dovi_tool -m 2 convert --discard -"
+	echo "mkvextract $1 tracks 0:${1%.*}.hevc"
 	echo "------------------"
-	if ! ffmpeg -i "$1" -c:v copy -vbsf hevc_mp4toannexb -f hevc - | dovi_tool -m 2 convert --discard -; then
-		echo "Failed to demux $1"
+	if ! mkvextract "$1" tracks 0:"${1%.*}.hevc"; then
```

Comment:
Nice, I was actually going to suggest this same thing. Switching to `mkvextract` removes the direct dependency on `ffmpeg`. The only small downside is that the video track has to be first (track 0) with this implementation (because `mkvextract` expects a track order specifier). I think that's fine as the video track will be almost always be track 0.